### DISCUSSION TITLE
TimeStepper System

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [idaholab/moose#23729] [Q&A General] [answered]
What is the default time solving system in moose?
I see that the introduction of BDF2 only has the first time derivative format, so what method do I use for the second time derivative term?
How do I view the methods used?
Please tell me, thank you very much!

**Comment by GiudGiud** [comment, marked as answer]
Hello

First some distinction we make is that there are both:
- a TimeStepper (by default created from end_time and num_steps to be constant), which handles the size of the steps
- a TimeIntegrator (by default implicit euler) which handles the time integration
All the methods may be found listed here:
https://mooseframework.inl.gov/moose/syntax/Executioner/TimeStepper/index.html
https://mooseframework.inl.gov/moose/syntax/Executioner/TimeIntegrator/index.html

There are dotdot() routines to get double derivatives, for example in the Coupleable interface
https://mooseframework.inl.gov/docs/doxygen/moose/classCoupleable.html#abf460b3c31b68245a81855f1656bfa43

Guillaume

**Reply by nanguaxiaofendui** [reply]
thank you
